testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -728,36 +730,51 @@ public void testRenameTableAcrossSchema()
     }
 
     @Test
-    public void testDeleteAllTable()
+    public void verifySupportsDelete()
     {
-        if (!hasBehavior(SUPPORTS_DELETE)) {
-            assertQueryFails("DELETE FROM region", "This connector does not support deletes");
+        if (hasBehavior(SUPPORTS_DELETE)) {
+            // Covered by testDelete*
             return;
         }
 
-        String tableName = "test_delete_" + randomTableSuffix();
-        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM nation", 25);
-        // not using assertUpdate as some connectors provide update count and some not
-        getQueryRunner().execute("DELETE FROM " + tableName);
-        assertQuery("SELECT count(*) FROM " + tableName, "VALUES 0");
-        assertUpdate("DROP TABLE " + tableName);
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "AS SELECT * FROM nation")) {
+            assertQueryFails("DELETE FROM " + table.getName(), "This connector does not support deletes");
+        }
     }
 
     @Test
-    public void testRowLevelDelete()
+    public void verifySupportsRowLevelDelete()
     {
-        if (!hasBehavior(SUPPORTS_DELETE)) {
-            assertQueryFails("DELETE FROM region", "This connector does not support deletes");
+        if (hasBehavior(SUPPORTS_ROW_LEVEL_DELETE)) {
+            // Covered by testRowLevelDelete
             return;
         }
-        if (!hasBehavior(SUPPORTS_ROW_LEVEL_DELETE)) {
-            assertQueryFails("DELETE FROM nation WHERE nationkey = 2", "This connector does not support deletes");
-            return;
+
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA));
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "AS SELECT * FROM nation")) {
+            assertQueryFails("DELETE FROM " + table.getName() + " WHERE nationkey = 2", "This connector does not support deletes");
+        }
+    }
+
+    @Test
+    public void testDeleteAllTable()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE_WITH_DATA) && hasBehavior(SUPPORTS_DELETE));
```

Comment:
Do we need `SUPPORTS_CREATE_TABLE_WITH_DATA`? I know test in this shape requires that, but I would rather have test fail for some connector which supports DELETE but not CREATE_TABLE_WITH_DATA; so we know we need to write it up differently for this connector, instead having silent skip and no testing for delete functinality.

Reply:
Fair point. Tests can be rewritten to use maybe CREATE+INSERT or some INSERT via remote database in subclasses.